packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -214,7 +212,6 @@ class SocksProxyConnection {
         });
 
         internalTLS.once('end', () => closeBothSockets());
-        targetTLS.once('end', () => closeBothSockets());
```

Comment:
The order is the following:

- `targetTLS` `'end'` event
- `targetTLS` `'error'` event (we are interested in that and want to write to the `internalTLS` there but the problem is that `internalTLS` gets closed in the previous `'end'` event.)

Reply:
It's save to not listen on `targetTLS.once('close` here since we already to it in line 94.